handler/src/main/java/io/netty/handler/ssl/SslContextMapping.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright 2017 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.handler.ssl;
+
+import io.netty.util.AsyncMapping;
+import io.netty.util.Supplier;
+
+/**
+ * A map of hostnames to {@link Supplier}s of {@link SslContext}s.
+ *
+ * @see SniHandler
+ */
+public interface SslContextMapping extends AsyncMapping<String, Supplier<? extends SslContext>> {
```

Comment:
That would be the thought but I think the interface should be this instead:

```java
interface SslContextMapping<T extends SslContext> extends AsyncMapping<String, T> {}
```

From Netty's POV we don't gain or lose anything if it's just SslContext. From the user's POV we lose the ability to clamp down on the generics and facilitate the compiler for type/input validation.